overlord/snapstate/handlers.go

```
@@ -579,7 +585,8 @@ func (m *SnapManager) undoUnlinkCurrentSnap(t *state.Task, _ *tomb.Tomb) error {
 	}
 
 	snapst.Active = true
-	err = m.backend.LinkSnap(oldInfo)
+	model, _ := Model(st)
```

we shouldn't ignore the error when is not ErrNoState I think